Doc/c-api/structures.rst

```diff
@@ -70,6 +70,17 @@ the definition of all other Python objects.
       (((PyObject*)(o))->ob_type)
 
 
+.. c:macro:: Py_IS_TYPE(o, type)
+
+   This macro is used to check whether the type of object *o* is the *type*.
+   It expands to::
+
+      (((PyObject*)(o))->ob_type) == (type))
```

Comment:
I suggest:

Check whether the type of object *o* is equal to *type*. Equivalent to `Py_TYPE(o) == type`.

I dislike documentating the exact implementation.